src/docs/preferences.md

```diff
@@ -4,191 +4,522 @@ title: Preferences
-
-As of right now the files must contain valid JSON containing the names and values of preferences (note that the following preference names are not official and only used as an example). You can also add comments to the settings.json file if needed i.e.
-
-```
+Theia provides a comprehensive preferences system that allows extensions to contribute configuration options, manage default values, and respond to preference changes. The system supports both frontend and backend environments and offers multiple configuration scopes.
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Preference Scopes](#preference-scopes)
+- [Preference Files](#preference-files)
+- [Contributing Preferences](#contributing-preferences)
+  - [1. Define a Preference Schema](#1-define-a-preference-schema)
+  - [2. Create a Configuration Interface (Optional but Recommended)](#2-create-a-configuration-interface-optional-but-recommended)
+  - [3. Create a Preference Proxy (Optional)](#3-create-a-preference-proxy-optional)
+  - [4. Register Your Preferences](#4-register-your-preferences)
+  - [Understanding the `scope` Property](#understanding-the-scope-property)
+- [Using Preferences](#using-preferences)
+  - [Minimal Approach: Direct Access via Preference Service](#minimal-approach-direct-access-via-preference-service)
+  - [Type-Safe Access via Preference Proxy](#type-safe-access-via-preference-proxy)
+- [Advanced Features](#advanced-features)
+  - [Language-Specific Preferences](#language-specific-preferences)
+  - [Preference Inspection](#preference-inspection)
+  - [Resource-Specific Preferences](#resource-specific-preferences)
+  - [Preference Overrides and Defaults](#preference-overrides-and-defaults)
+- [Backend Preferences](#backend-preferences)
+  - [Key Points for Backend Usage](#key-points-for-backend-usage)
+  - [Backend Usage Example](#backend-usage-example)
+  - [Creating a Backend Preference Service](#creating-a-backend-preference-service)
+- [Architecture Notes](#architecture-notes)
+- [Best Practices](#best-practices)
+- [Troubleshooting](#troubleshooting)
+  - [Common Issues](#common-issues)
+  - [Debugging](#debugging)
+- [Examples](#examples)
+
+## Overview
+
+The Theia preferences system consists of several key components:
+
+- **Preference Schema**: Defines the structure and metadata of preferences
+- **Preference Service**: Core service for reading and writing preference values
+- **Preference Proxies**: Type-safe interfaces for accessing preferences
+- **Preference Scopes**: Different levels where preferences can be stored
+
+## Preference Scopes
+
+Preferences in Theia are organized into scopes from most general to most specific:
+
+1. **Default** - Built-in default values defined by extensions
+2. **User** - Global user preferences stored in `~/.theia/settings.json`
+3. **Workspace** - Workspace-specific preferences stored in `<workspace>/.theia/settings.json`
```

Comment:
For multi-folder workspaces, the preferences can also be stored in the workspace description file.